connector/web3/web3.go

```diff
@@ -48,6 +50,44 @@ func (c *web3Connector) InfuraID() string {
 
 var erc1271magicValue = [4]byte{0x16, 0x26, 0xba, 0x7e}
 
+// https://gist.github.com/dcb9/385631846097e1f59e3cba3b1d42f3ed#file-eth_sign_verify-go
+func (c *web3Connector) Valid(address, nonce, redirectUri string, message siwe.Message, signature string) (identity connector.Identity, err error) {
+	if message.GetAddress() != common.HexToAddress(address) {
+		return identity, fmt.Errorf("signed message address %s doesn't match the original nonce address %s", message.GetAddress().String(), common.HexToAddress(address).String())
+	}
+
+	if message.GetNonce() != nonce {
+		return identity, fmt.Errorf("signed message nonce %s doesn't match the original nonce %s", message.GetNonce(), nonce)
+	}
+
+	redirect, err := url.Parse(redirectUri)
+	if err != nil {
+		return identity, fmt.Errorf("signed message has invalid URI: %v", err)
+	}
+
+	if message.GetDomain() != redirect.Host {
+		return identity, fmt.Errorf("signed message domain %s doesn't match the original domain %s", message.GetDomain(), redirect.Host)
+	}
+
+	if message.GetURI() != *redirect {
```

Comment:
This is a little suspicious. I'm not sure comparing these big structs will work. The auth request in the DB has a redirect URI in it, doesn't it?